packages/jellyfish-api-core/__tests__/category/spv/refundHtlc.test.ts

```diff
@@ -0,0 +1,104 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { BigNumber, RpcApiError } from '@defichain/jellyfish-api-core'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+describe('Spv', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+
+    await container.spv.fundAddress(await container.call('spv_getnewaddress')) // Funds 1 BTC
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should refundHtlc', async () => {
+    const pubKeyA = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const pubKeyB = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const htlc = await container.call('spv_createhtlc', [pubKeyA, pubKeyB, '10'])
+
+    await container.call('spv_sendtoaddress', [htlc.address, 0.1]) // Funds HTLC address
+
+    const destinationAddress = await container.call('spv_getnewaddress')
+    const result = await client.spv.refundHtlc(htlc.address, destinationAddress)
```

Comment:
This seems incorrect with the real-life scenario. The Buyer has to wait until a timeout to call a  refund.
refer https://en.bitcoin.it/wiki/BIP_0199
But we experience this behavior because in `regtest` we do not connect to the Bitcoin network, rather fake it. either these time constraints should be added to C++ side spv fake wrapper. But for now please add a comment, otherwise a reader could get the wrong idea.